frontend/packages/core/src/NPS/feedback.tsx

```diff
@@ -0,0 +1,219 @@
+import React, { useState } from "react";
+import { clutch as IClutch } from "@clutch-sh/api";
+import styled from "@emotion/styled";
+import { Grid as MuiGrid } from "@material-ui/core";
+import MuiSuccessIcon from "@material-ui/icons/CheckCircle";
+import { debounce } from "lodash";
+import { v4 as uuid } from "uuid";
+
+import { userId } from "../AppLayout/user";
+import { Button } from "../button";
+import { Alert } from "../Feedback";
+import { TextField } from "../Input";
+import { client } from "../Network";
+import type { ClutchError } from "../Network/errors";
+import { Typography } from "../typography";
+
+import EmojiRatings, { Rating } from "./emojiRatings";
+
+/** Interfaces */
+
+interface FeedbackOptions {
+  origin: "ORIGIN_UNSPECIFIED" | "WIZARD" | "ANYTIME";
+  onSubmit?: (submit: boolean) => void;
+}
+
+// Defaults in case of API failure
+export const defaults: IClutch.feedback.v1.ISurvey = {
+  prompt: "Rate Your Experience",
+  freeformPrompt: "What would you recommend to improve this?",
+  ratingLabels: [
+    {
+      emoji: 1,
+      label: "bad",
+    },
+    {
+      emoji: 2,
+      label: "ok",
+    },
+    {
+      emoji: 3,
+      label: "great",
+    },
+  ],
+};
+
+const StyledButton = styled(Button)({
+  fontSize: "14px",
+  padding: "0 8px",
+  height: "18px",
+});
+
+const StyledTextField = styled(TextField)({
+  marginTop: "15px",
+  ".MuiInputBase-root": {
+    fontSize: "14px",
+  },
+});
+
+/**
+ * NPS feedback component which is the base for both Wizard and Anytime
+ * Will fetch given survey options from the server based on the given origin
+ * Then display a feedback component based on the given emoji ratings
+ *
+ * @param opts Available feedback options
+ * @returns NPSFeedback component
+ */
+const NPSFeedback = (opts: FeedbackOptions = { origin: "ORIGIN_UNSPECIFIED" }) => {
```

Comment:
is it okay if a default isn't set?
```suggestion
const NPSFeedback = (opts: FeedbackOptions ) => {
```

Reply:
Should be, I'm not exporting the NPSFeedback component intentionally so it will only be used via the NPSWizard and NPSAnytime components at the moment. Once we get the Anytime component in I will probably make that the default though.